seumich/tests.py

```diff
@@ -525,6 +528,46 @@ def test_advisor_view(self):
         self.assertContains(response, 'james')
         self.assertNotContains(response, 'grace')
 
+    def test_advisor_view_user_without_mentor(self):
+        # Set up
+        user_ebenezer = get_user_model().objects.create(
+            username="ebenezer", first_name="Ebenezer", last_name="Scrooge"
+        )
+        user_ebenezer.set_password("ebenezer")
+        user_ebenezer.save()
+        # Test
+        self.client.login(username="ebenezer", password="ebenezer")
+        url = reverse("seumich:advisor", kwargs={"advisor": "ebenezer"})
+        response = self.client.get(url)
+        full_name = response.context["studentListHeader"]
+        self.assertEqual(full_name, "Ebenezer Scrooge")
+        self.assertEqual(len(response.context["students"]), 0)
+        # Clean up
+        user_ebenezer.delete()
+
+    def test_advisor_view_mentor_without_user(self):
+        # Set up
+        expected_students = [
+            {'username': 'james', 'first_name': 'James', 'last_name': 'Bond'},
+            {'username': 'gianna', 'first_name': 'Gianna', 'last_name': 'Fekete'},
+            {'username': 'deirdre', 'first_name': 'Deirdre', 'last_name': 'Haupt'},
+            {'username': 'gabriela', 'first_name': 'Gabriela', 'last_name': 'Rea'}
+        ]
+        user_lavera = get_user_model().objects.get(username="lavera")
+        user_lavera.delete()
+        # Test
+        self.client.login(username='burl', password='burl')
+        url = reverse('seumich:advisor', kwargs={'advisor': 'lavera'})
+        response = self.client.get(url)
+        students = list(response.context['students'].values('username', 'first_name', 'last_name'))
+        self.assertCountEqual(expected_students, students)
+        # Clean up
+        user_lavera = get_user_model().objects.create(
```

Comment:
You don't need to re-create the object, you can just save what you retrieved back on line 556. Just remove lines 565-568 and this will work the same.

Reply:
That makes sense! Change made, see [latest commit](https://github.com/tl-its-umich-edu/student_explorer/pull/168/commits/5f4dd402abbb1e621c0c6831f41933feb0eabde0).